catalyst/core/callbacks/pruning.py

```diff
@@ -0,0 +1,144 @@
+        prune_on_epoch_end: bool = False,
+        prune_on_stage_end: bool = True,
+        remove_reparametrization: bool = True,
+        reinitialize_after_pruning: bool = False,
+        layers_to_prune: Optional[List[str]] = None,
+    ) -> None:
+        """
+        Init method for pruning callback
+
+        Args:
+            pruner_fn: function from torch.nn.utils.prune module
+                or your based on BasePruningMethod. See pytorch
+                docs for more details.
+            key_to_prune: can be string or list of strings. Determines
+                which tensor in modules will be pruned.
+            amount: quantity of parameters to prune.
+                If float, should be between 0.0 and 1.0 and
+                represent the fraction of parameters to prune.
+                If int, it represents the absolute number
+                of parameters to prune.
+            prune_on_epoch_end: bool flag determines call or not
+                to call pruning_fn on epoch end.
+            prune_on_stage_end: bool flag determines call or not
+                to call pruning_fn on stage end.
+            remove_reparametrization: if True then all reparametrization
+                pre-hooks and tensors with mask will be removed on
+                stage end.
+            reinitialize_after_pruning: if True then will reinitialize model
+                after pruning. (Lottery Ticket Hypothesis)
+            layers_to_prune: list of strings - module names to be pruned.
+                If None provided then will try to prune every module in
+                model.
+
+        """
+        super().__init__(CallbackOrder.External)
+        self.pruner_fn = pruner_fn
+        self.prune_on_epoch_end = prune_on_epoch_end
+        self.prune_on_stage_end = prune_on_stage_end
+        if not (prune_on_stage_end or prune_on_epoch_end):
+            warnings.warn(
+                "Warning!"
+                "You disabled pruning pruning both on epoch and stage end."
+                "Model won't be pruned by this callback."
+            )
+        self.remove_reparametrization = remove_reparametrization
+        self.key_to_prune = key_to_prune
+        self.amount = amount
+        self.reinitialize_after_pruning = reinitialize_after_pruning
+        self.layers_to_prune = layers_to_prune
+
+    @staticmethod
+    def _weight_reset(m):
+        try:
+            m.reset_parameters()
+        except AttributeError:
+            pass
+
+    def _prune_module(self, module):
+        if isinstance(self.key_to_prune, str):
+            self.pruner_fn(module, name=self.key_to_prune, amount=self.amount)
+        elif isinstance(self.key_to_prune, list):
+            for key in self.key_to_prune:
+                self.pruner_fn(module, name=key, amount=self.amount)
+
+    def _to_be_pruned(self, layer_name):
+        if self.layers_to_prune is None:
+            return True
+        return layer_name in self.layers_to_prune
```

Comment:
could we make it `self.layers_to_prune is None or layer_name in self.layers_to_prune`?